splink/spark/spark_linker.py

```diff
@@ -140,10 +142,18 @@ def __init__(
                 " argument when you initialise the linker"
             )
 
-        for table in self.spark.catalog.listTables():
-            if table.isTemporary:
-                if "__splink__" in table.name:
-                    self.spark.catalog.dropTempView(table.name)
+        # set the catalog and database of where to write output tables
+        self.catalog = catalog if catalog is not None else spark.catalog.currentCatalog()
+        self.database = database if database is not None else spark.catalog.currentDatabase()
+
+        # if we use spark.sql("USE DATABASE db") commands we change the default. This approach prevents side effects.
+        splink_tables = spark.sql(f"show tables from {self.catalog}.{self.database} like '*__splink__*'")
```

Comment:
This is not relevant to this PR, but while I'm thinking about it Re: your finding about `__splink__df_concat_with_tf` not changing if a second linker is instantiated.  

This (the existing implementation, but probably your new one as well) is the source of the bug.  It looks to me like the bug will occur only if two linkers are in play simultaneously.  If you create a new linker, it looks (to me at least from this code, I haven't tested), that it should correctly destroy any existing tables called `__splink__df_concat_with_tf `

Reply:
I'll try and replicate, but I did notice that the issue was in dropping temp tables only - there was another persistent table that remained. This was the culprit - __splink__input_table_0

I'm also not sure if this is because I was changing the input data, but not the settings dict. I'll keep an eye and see if I can get a more concrete description of the behaviour

Reply:
Yep - that sounds very plausible as the source of the problem